tests/cluster_test.go

```diff
@@ -0,0 +1,88 @@
+package k3k_test
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/rancher/k3k/pkg/apis/k3k.io/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+)
+
+var _ = When("a cluster is installed", func() {
+
+	var namespace string
+
+	BeforeEach(func() {
+		createdNS := &corev1.Namespace{ObjectMeta: v1.ObjectMeta{GenerateName: "ns-"}}
+		createdNS, err := k8s.CoreV1().Namespaces().Create(context.Background(), createdNS, v1.CreateOptions{})
+		Expect(err).To(Not(HaveOccurred()))
+		namespace = createdNS.Name
+	})
+
+	It("will be created in shared mode", func() {
+		cluster := v1alpha1.Cluster{
+			ObjectMeta: v1.ObjectMeta{
+				Name:      "mycluster",
+				Namespace: namespace,
+			},
+			Spec: v1alpha1.ClusterSpec{
+				Mode:    v1alpha1.SharedClusterMode,
+				Servers: ptr.To[int32](1),
+				Agents:  ptr.To[int32](0),
+				Version: "v1.26.1-k3s1",
+			},
+		}
+
+		err := k8sClient.Create(context.Background(), &cluster)
+		Expect(err).To(Not(HaveOccurred()))
+
+		By("checking server and kubelet readiness state")
+
+		// check that the server Pod and the Kubelet are in Ready state
+		Eventually(func() bool {
+			podList, err := k8s.CoreV1().Pods(namespace).List(context.Background(), v1.ListOptions{})
+			Expect(err).To(Not(HaveOccurred()))
+
+			serverRunning := false
+			kubeletRunning := false
+
+			for _, pod := range podList.Items {
+				imageName := pod.Spec.Containers[0].Image
+				imageName = strings.Split(imageName, ":")[0] // remove tag
+
+				switch imageName {
+				case "rancher/k3s":
+					serverRunning = pod.Status.Phase == corev1.PodRunning
+				case "rancher/k3k-kubelet":
+					kubeletRunning = pod.Status.Phase == corev1.PodRunning
+				}
+
+				if serverRunning && kubeletRunning {
+					return true
+				}
+			}
+
+			return false
```

Comment:
Apologies for the noob question as I'm not super familiar with go test :) 
If it's false , do we want to give more context (e.g logs?)

Reply:
Probably not in this case. This is a continuous loop that will check for the readiness state, until the timeout occurs.

If the timed out we know that the pods were not Ready, and at the end of the tests the cluster logs will be available to download: https://github.com/rancher/k3k/actions/runs/12827942200/artifacts/2446381844

Those are very convenient to check why the tests failed!

![image](https://github.com/user-attachments/assets/631fc978-1b8e-4796-9e14-82f016268b10)